src/onegov/org/models/ticket.py

```diff
@@ -366,15 +337,25 @@ def resource(self):
         return query.one()
 
     @cached_property
-    def reservations(self):
+    def reservations_query(self):
         # libres allows for multiple reservations with a single request (token)
         # for now we don't really have that case in onegov.org, but we
         # try to be aware of it as much as possible
         query = self.session.query(Reservation)
         query = query.filter(Reservation.token == self.id)
         query = query.order_by(Reservation.start)
+        return query
 
-        return tuple(query)
+    @cached_property
+    def reservations(self):
+        return tuple(self.reservations_query)
+
+    def has_future_reservation(self):
```

Comment:
Maybe we should make this a `cached_property`, unless you're confident we only call this once per request. This will also mean changing all the calls to simple attribute accesses.